rpcs3/rpcs3qt/vfs_dialog_path_widget.cpp

```diff
@@ -163,18 +184,25 @@ void vfs_dialog_path_widget::update_selection()
 
 			if (is_selected)
 			{
-				m_dir_list->setCurrentItem(item);
+				if (m_dir_list->currentRow() == -1) // If no current row is set (e.g. at startup)
+				{
+					m_dir_list->setCurrentRow(i); // Set both the new current item and new current row
+				}
+				else if (m_dir_list->currentRow() != i) // If current row != checked row, make sure to enable the row
+				{
+					m_dir_list->currentRowChanged(m_dir_list->currentRow());
```

Comment:
Why are you sending a base class signal here?
If you only want the code in your connect, then make it a function and call it directly.
And why does it set the current row again?

Reply:
this is invoked even when you click on the small checkbox area to simply change the current path and not the row. But in this case you make sure that the row get focus and the  `-` button is enabled (if row > 0). These are made also on currentRowChanged. I reused that method instead of duplicating it (I would need also to pass the reference  `button_remove_dir`)

Reply:
what i mean is: make a slot for the stuff you do in that connect and call it instead of sending the base class signal

Reply:
Simplified the logic relying on stylesheet instead of focus. So that old logic is no more present